dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/AsyncContextImpl.java

```diff
@@ -34,10 +34,6 @@
     private RpcContext storedServerContext;
 
     public AsyncContextImpl() {
-    }
-
-    public AsyncContextImpl(CompletableFuture<Object> future) {
-        this.future = future;
         this.storedContext = RpcContext.getContext();
```

Comment:
as RpcContext.getContext() and RpcContext.getServerContext() is are per thread specific and can be available to caller using based on thread, so should we store them as separate member variable of this class? What do you say?

Reply:
```java
    public AsyncContextImpl() {
        this.storedContext = RpcContext.getContext();
        this.storedServerContext = RpcContext.getServerContext();
    }
```